test/binding-redirects.Tests/BindingRedirectTests.cs

```diff
@@ -0,0 +1,286 @@
+            var runtimeSectionXml = config.Sections["runtime"].SectionInformation.GetRawXml();
+            var runtimeSectionElement = XElement.Parse(runtimeSectionXml);
+            var redirects = runtimeSectionElement.Elements()
+                                .Where(e => e.Name.LocalName == "assemblyBinding").Elements()
+                                .Where(e => e.Name.LocalName == "dependentAssembly");
+            return redirects;
+        }
+
+        private void VerifyRedirects(IEnumerable<XElement> redirects, IEnumerable<XElement> generatedBindings)
+        {
+            foreach (var binding in generatedBindings)
+            {
+                var redirect = redirects.SingleOrDefault(r => /*XNode.DeepEquals(r, binding)*/ r.ToString() == binding.ToString());
+
+                redirect.Should().NotBeNull($"Binding not found in runtime section : {Environment.NewLine}{binding}");
+            }
+        }
+
+        private void VerifyAppSettings(string exePath)
+        {
+            var configFile = ConfigurationManager.OpenExeConfiguration(exePath);
+            foreach (var appSetting in ExpectedAppSettings)
+            {
+                var value = configFile.AppSettings.Settings[appSetting.Key];
+                value.Should().NotBeNull($"AppSetting with key '{appSetting.Key}' not found in config file.");
+                value.Value.Should().Be(appSetting.Value, $"For AppSetting '{appSetting.Key}' - Expected Value '{appSetting.Value}', Actual '{ value.Value}'");
+            }
+        }
+
+        [WindowsOnlyFact]
+        public void Build_Generates_Redirects_For_App_Without_Config()
+        {
+            var redirects = GetRedirects(_appWithoutConfigBuildOutput);
+            VerifyRedirects(redirects, ExpectedBindingsAppNoConfig);
+
+            var commandResult = new TestCommand(_appWithoutConfigBuildOutput)
+                                    .Execute();
+            commandResult.Should().Pass();
+        }
+
+        [WindowsOnlyFact]
+        public void Publish_Generates_Redirects_For_App_Without_Config()
+        {
+            var redirects = GetRedirects(_appWithoutConfigPublishOutput);
+            VerifyRedirects(redirects, ExpectedBindingsAppNoConfig);
+
+            var commandResult = new TestCommand(_appWithoutConfigPublishOutput)
+                                    .Execute();
+            commandResult.Should().Pass();
+        }
+
+        [WindowsOnlyFact]
+        public void Build_Generates_Redirects_For_Executable_Dependency()
+        {
+            var redirects = GetRedirects(_executableDependencyBuildOutput);
+            VerifyRedirects(redirects, ExpectedBindingsAppNoConfig);
+
+            var commandResult = new TestCommand(_executableDependencyBuildOutput)
+                                    .Execute();
+            commandResult.Should().Pass();
+        }
+
+        //[WindowsOnlyFact]
+        public void Publish_Generates_Redirects_For_Executable_Dependency()
```

Comment:
@NTaylorMullen @anurse @pakrym @davidfowl I like to hear your thoughts on this scenario. Tools package is generally a development scenario. Currently (with this fix) `dotnet build` generates binding redirects for any packages containing executables. Do you think that `dotnet publish` (which is a production scenario) must also support generate binding redirects for executable dependencies?